codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/AcceptorGenerator.kt

```diff
@@ -89,12 +89,12 @@ private fun KotlinWriter.renderPathAcceptor(wi: WaiterInfo, directive: String, i
         val comparison = when (matcher.comparator!!) {
             PathComparator.STRING_EQUALS -> "${actual.identifier} == ${expected.dq()}"
             PathComparator.BOOLEAN_EQUALS -> "${actual.identifier} == ${expected.toBoolean()}"
-            PathComparator.ANY_STRING_EQUALS -> "${actual.identifier}?.any { it == ${expected.dq()} } ?: false"
+            PathComparator.ANY_STRING_EQUALS -> "(${actual.identifier} as List<String>?)?.any { it == ${expected.dq()} } ?: false"
 
             // NOTE: the isNotEmpty check is necessary because the waiter spec says that `allStringEquals` requires
             // at least one value unlike Kotlin's `all` which returns true if the collection is empty
             PathComparator.ALL_STRING_EQUALS ->
-                "!${actual.identifier}.isNullOrEmpty() && ${actual.identifier}.all { it == ${expected.dq()} }"
+                "!(${actual.identifier} as List<String>).isNullOrEmpty() && ${actual.identifier}.all { it == ${expected.dq()} }"
         }
```

Comment:
**Question**: Why is this new cast to `List<String>` necessary?

Reply:
Long explanation but when multi select hash makes a selection that selection is stored in an object. Something like this: `class Selection<T>(val x: T, val y: T)`. 

The `.isNullOrEmpty()` function can only be applied to arrays, lists and such. 

So at runtime when it was detecting the selection properties as generic (x or y) I was getting errors (because x or y didn't show up as a list, array, etc), that cast fixed them.